Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,130 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023-2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import CoreCommands
+import TSCUtility
+import TSCBasic
+import PackageModel
+import Foundation
+
+extension SwiftPackageTool {
+    
+    struct Install: SwiftCommand {
+        static var configuration: CommandConfiguration {
+            CommandConfiguration(abstract: "Offers the ability to install executable targets of the current package.")
+        }
+        
+        @OptionGroup()
+        var globalOptions: GlobalOptions
+        
+        func run(_ tool: SwiftTool) throws {
+            let dotSwiftPMDir = try tool.fileSystem.getOrCreateDotSwiftPMDirectory()
+            
+            let env = ProcessInfo.processInfo.environment
+            
+
+            // ~/.swiftpm/bin/
+            let swiftpmBinDir = dotSwiftPMDir.appending(component: "bin")
+            
+            if let path = env.path, !path.contains(swiftpmBinDir.pathString), env["SWIFTPM_SHUT_UP_ABOUT_EXECUTABLE_PATH"] == nil {
+                tool.observabilityScope.emit(warning: "PATH doesn't include \(swiftpmBinDir.pathString)! This means you won't be able to access the installed executables by default, and will need to specify the full path. If you wish to stop seeing this warning, set the enviroment variable `SWIFTPM_SHUT_UP_ABOUT_EXECUTABLE_PATH`.")
+            }
+            
+            let registryURL = dotSwiftPMDir.appending(component: "installedPackages.json").asURL
+            
+            var alreadyExisting = (try? InstalledPackage.registered(registryURL: registryURL)) ?? []
+            
+            let workspace = try tool.getActiveWorkspace()
+            let packageRoot = try tool.getPackageRoot()
+
+            let packageGraph = try tsc_await {
```

Comment:
`tsc_await` is going to be deprecated and replaced with the use of actual `await`. To not increase the reliance on `tsc_await` you could wrap `loadRootPackage` in an `async` function that passes continuation's `resume` method as the `completion` argument and then you can call that here with `await` by adding `async` effect on `run` and changing protocol conformance from `SwiftCommand` to `AsyncSwiftCommand`.